lib/commands/actions.js

```diff
@@ -250,11 +250,19 @@ helpers.getScreenshotDataWithAdbShell = async function (adb, opts) {
 };
 
 helpers.getScreenshotDataWithAdbExecOut = async function (adb) {
-  let {stdout} = await exec(adb.executable.path, ['exec-out', '/system/bin/screencap -p'],
-                            {encoding: 'binary', isBuffer: true});
+  let {stdout, stderr, code} = await exec(adb.executable.path,
+                                    adb.executable.defaultArgs
+                                      .concat(['exec-out', '/system/bin/screencap', '-p']),
+                                    {encoding: 'binary', isBuffer: true});
+  // if there is an error, throw
+  if (code || stderr.length) {
+    throw new Error(`Screenshot returned error, code: '${code}', stderr: '${stderr.toString()}'`);
+  }
+  // if we don't get anything at all, throw
   if (!stdout.length) {
-    throw new Error('The size of the taken screenshot equals to zero.');
+    throw new Error('Screenshot returned no data');
```

Comment:
I'd also add an explanation text here like "Make sure the LayoutParams.FLAG_SECURE is not set for the current view". There is also the second place where this error might be show below in the _getScreenshotDataWithAdbShell_ method

Reply:
That message can be handled a little lower. And the error handling for the `adb shell` path is handled in `appium-adb`. Since that is bypassed for `exec-out` it needs to be handled here.